<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Internal Function to Get Mutations/CNA/Fusion By Sample ID — .get_data_by_sample • cbioportalR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Internal Function to Get Mutations/CNA/Fusion By Sample ID — .get_data_by_sample"><meta name="description" content="Internal Function to Get Mutations/CNA/Fusion By Sample ID"><meta property="og:description" content="Internal Function to Get Mutations/CNA/Fusion By Sample ID"><meta property="og:image" content="https://karissawhiting.github.io/cbioportalR/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZFP6P2RCRE"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-ZFP6P2RCRE');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">cbioportalR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/overview-of-workflow.html">Overview of Data Retrieval Workflow</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/karissawhiting/cbioportalR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Internal Function to Get Mutations/CNA/Fusion By Sample ID</h1>
      <small class="dont-index">Source: <a href="https://github.com/karissawhiting/cbioportalR/blob/v1.1.1/R/genomics_by_sample.R" class="external-link"><code>R/genomics_by_sample.R</code></a></small>
      <div class="d-none name"><code>dot-get_data_by_sample.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Internal Function to Get Mutations/CNA/Fusion By Sample ID</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">.get_data_by_sample</span><span class="op">(</span></span>
<span>  sample_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  study_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  molecular_profile_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sample_study_pairs <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mutation"</span>, <span class="st">"cna"</span>, <span class="st">"fusion"</span>, <span class="st">"structural_variant"</span>, <span class="st">"segment"</span><span class="op">)</span>,</span>
<span>  genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  panel <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  add_hugo <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  base_url <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-sample-id">sample_id<a class="anchor" aria-label="anchor" href="#arg-sample-id"></a></dt>
<dd><p>a vector of sample IDs (character)</p></dd>


<dt id="arg-study-id">study_id<a class="anchor" aria-label="anchor" href="#arg-study-id"></a></dt>
<dd><p>A string indicating the study ID from which to pull data. If no study ID, will
guess the study ID based on your URL and inform. Only 1 study ID can be passed. If mutations/cna from
more than 1 study needed, see <code>sample_study_pairs</code></p></dd>


<dt id="arg-molecular-profile-id">molecular_profile_id<a class="anchor" aria-label="anchor" href="#arg-molecular-profile-id"></a></dt>
<dd><p>A string indicating the molecular profile ID from which to pull data. If ID supplied, will
guess the molecular profile ID based on the study ID. Only 1 molecular profile ID can be passed. If mutations from
more than 1 study needed, see <code>sample_study_pairs</code></p></dd>


<dt id="arg-sample-study-pairs">sample_study_pairs<a class="anchor" aria-label="anchor" href="#arg-sample-study-pairs"></a></dt>
<dd><p>A dataframe with columns: <code>sample_id</code>, <code>study_id</code> and <code>molecular_profile_id</code> (optional). Variations in capitalization of column names are accepted.
This can be used in place of <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> arguments above if you
need to pull samples from several different studies at once. If passed this will take overwrite <code>sample_id</code>, <code>study_id</code>, <code>molecular_profile_id</code> if also passed.</p></dd>


<dt id="arg-data-type">data_type<a class="anchor" aria-label="anchor" href="#arg-data-type"></a></dt>
<dd><p>specify what type of data to return. Options are<code>mutations</code>, <code>cna</code>, <code>fusion</code> or <code>structural_variant</code> (same as <code>fusion</code>).</p></dd>


<dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p>A vector of Entrez ids or Hugo symbols. If Hugo symbols are supplied, they will be converted to entrez ids using the <code><a href="get_entrez_id.html">get_entrez_id()</a></code> function.
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p></dd>


<dt id="arg-panel">panel<a class="anchor" aria-label="anchor" href="#arg-panel"></a></dt>
<dd><p>One or more panel IDs to query (e.g. 'IMPACT468').
If <code>panel</code> and <code>genes</code> are both supplied, genes from both arguments will be returned. If both are NULL (default), it will return gene results for all available genomic data for that sample.</p></dd>


<dt id="arg-add-hugo">add_hugo<a class="anchor" aria-label="anchor" href="#arg-add-hugo"></a></dt>
<dd><p>Logical indicating whether <code>HugoGeneSymbol</code> should be added to your resulting data frame, if not already present in raw API results.
Argument is <code>TRUE</code> by default. If <code>FALSE</code>, results will be returned as is (i.e. any existing Hugo Symbol columns in raw results will not be removed).</p></dd>


<dt id="arg-base-url">base_url<a class="anchor" aria-label="anchor" href="#arg-base-url"></a></dt>
<dd><p>The database URL to query
If <code>NULL</code> will default to URL set with <code>set_cbioportal_db(&lt;your_db&gt;)</code></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe of mutations, CNAs or fusions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="set_cbioportal_db.html">set_cbioportal_db</a></span><span class="op">(</span><span class="st">"public"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> You are successfully connected!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">base_url</span> for this R session is now set to <span style="color: #0000BB;">"www.cbioportal.org/api"</span> </span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TCGA-OR-A5J2-01"</span>,<span class="st">"TCGA-OR-A5J6-01"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span> study_id <span class="op">=</span> <span class="st">"acc_tcga"</span>, data_type <span class="op">=</span> <span class="st">"mutation"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"acc_tcga"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">"acc_tcga_mutations"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 173 × 28</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    hugoGeneSymbol entrezGeneId uniqueSampleKey                  uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ZFPM1                <span style="text-decoration: underline;">161</span>882 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ZNF787               <span style="text-decoration: underline;">126</span>208 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> PODXL                  <span style="text-decoration: underline;">5</span>420 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> CCDC102A              <span style="text-decoration: underline;">92</span>922 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> TVP23C               <span style="text-decoration: underline;">201</span>158 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ZNF628                <span style="text-decoration: underline;">89</span>887 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> TBP                    <span style="text-decoration: underline;">6</span>908 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> SEMA5B                <span style="text-decoration: underline;">54</span>437 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> CELSR2                 <span style="text-decoration: underline;">1</span>952 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> MUC5B                <span style="text-decoration: underline;">727</span>897 VENHQS1PUi1BNUoyLTAxOmFjY190Y2dh VENHQS1PUi1BNUo…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 163 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 24 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;, startPosition &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;, …</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DS-sig-010-P2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span> molecular_profile_id <span class="op">=</span> <span class="st">"blca_plasmacytoid_mskcc_2016_cna"</span>, data_type <span class="op">=</span> <span class="st">"cna"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016_cna"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> YAP1                  <span style="text-decoration: underline;">10</span>413 RFMtc2lnLTAxMC1QMjpibGNhX3BsYXNt… RFMtc2lnLTAxMDp…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> CD79B                   974 RFMtc2lnLTAxMC1QMjpibGNhX3BsYXNt… RFMtc2lnLTAxMDp…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 5 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, alteration &lt;int&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P-0002146-T01-IM3"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span> study_id <span class="op">=</span> <span class="st">"blca_plasmacytoid_mskcc_2016"</span>, data_type <span class="op">=</span> <span class="st">"fusion"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016_structural_variants"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 44</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   uniqueSampleKey uniquePatientKey molecularProfileId sampleId patientId studyId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> UC0wMDAyMTQ2LV… UC0wMDAyMTQ2OmJ… blca_plasmacytoid… P-00021… P-0002146 blca_p…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 38 more variables: site1EntrezGeneId &lt;int&gt;, site1HugoSymbol &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1EnsemblTranscriptId &lt;chr&gt;, site1Chromosome &lt;chr&gt;, site1Position &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1Contig &lt;chr&gt;, site1Region &lt;chr&gt;, site1RegionNumber &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1Description &lt;chr&gt;, site2EntrezGeneId &lt;int&gt;, site2HugoSymbol &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2EnsemblTranscriptId &lt;chr&gt;, site2Chromosome &lt;chr&gt;, site2Position &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2Contig &lt;chr&gt;, site2Region &lt;chr&gt;, site2RegionNumber &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2Description &lt;chr&gt;, site2EffectOnFrame &lt;chr&gt;, ncbiBuild &lt;chr&gt;, …</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">df_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="st">"sample_id"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"s_C_36924L_P001_d"</span>,</span></span>
<span class="r-in"><span><span class="st">"s_C_03LNU8_P001_d"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span><span class="st">"study_id"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prad_msk_2019"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span>, data_type <span class="op">=</span> <span class="st">"mutation"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 28</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> TP53                   <span style="text-decoration: underline;">7</span>157 c19DXzAzTE5VOF9QMDAxX2Q6cHJhZF9t… cF9DXzAzTE5VODp…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 24 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;, startPosition &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   chr &lt;chr&gt;, variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, …</span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span>, genes <span class="op">=</span> <span class="fl">7157</span>, data_type <span class="op">=</span> <span class="st">"mutation"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">7157</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 28</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> TP53                   <span style="text-decoration: underline;">7</span>157 c19DXzAzTE5VOF9QMDAxX2Q6cHJhZF9t… cF9DXzAzTE5VODp…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 24 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;, startPosition &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   chr &lt;chr&gt;, variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, …</span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span>, data_type <span class="op">=</span> <span class="st">"cna"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">prad_msk_2019_cna</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 9</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> PTEN                   <span style="text-decoration: underline;">5</span>728 c19DXzM2OTI0TF9QMDAxX2Q6cHJhZF9t… cF9DXzM2OTI0TDp…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 5 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, alteration &lt;int&gt;</span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span>, data_type <span class="op">=</span> <span class="st">"fusion"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">prad_msk_2019_structural_variants</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 0 × 0</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">df_pairs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="st">"sample_id"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P-0002146-T01-IM3"</span>, <span class="st">"s_C_CAUWT7_P001_d"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span> <span class="st">"study_id"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blca_plasmacytoid_mskcc_2016"</span>, <span class="st">"prad_msk_2019"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs2</span>, data_type <span class="op">=</span> <span class="st">"mutation"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_plasmacytoid_mskcc_2016_mutations</span> and</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 7 × 28</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> TERT                   <span style="text-decoration: underline;">7</span>015 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> NOTCH4                 <span style="text-decoration: underline;">4</span>855 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> TP53                   <span style="text-decoration: underline;">7</span>157 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> BLM                     641 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> TP53                   <span style="text-decoration: underline;">7</span>157 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> CDKN1A                 <span style="text-decoration: underline;">1</span>026 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">7</span> RB1                    <span style="text-decoration: underline;">5</span>925 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 24 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;, startPosition &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, chr &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;, …</span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs2</span>, genes <span class="op">=</span> <span class="fl">7157</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_plasmacytoid_mskcc_2016_mutations</span> and</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">7157</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 28</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hugoGeneSymbol entrezGeneId uniqueSampleKey                   uniquePatientKey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> TP53                   <span style="text-decoration: underline;">7</span>157 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> TP53                   <span style="text-decoration: underline;">7</span>157 UC0wMDAyMTQ2LVQwMS1JTTM6YmxjYV9w… UC0wMDAyMTQ2OmJ…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 24 more variables: molecularProfileId &lt;chr&gt;, sampleId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   patientId &lt;chr&gt;, studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;, startPosition &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   chr &lt;chr&gt;, variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, …</span></span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs2</span>, data_type <span class="op">=</span> <span class="st">"cna"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_plasmacytoid_mskcc_2016_cna</span> and <span style="color: #0000BB;">prad_msk_2019_cna</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="fu">.get_data_by_sample</span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs2</span>, data_type <span class="op">=</span> <span class="st">"fusion"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining with `by = join_by(study_id)`</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following parameters were used in query:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_plasmacytoid_mskcc_2016"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_plasmacytoid_mskcc_2016_structural_variants</span> and</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">prad_msk_2019_structural_variants</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"All available genes"</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 44</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   uniqueSampleKey uniquePatientKey molecularProfileId sampleId patientId studyId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> UC0wMDAyMTQ2LV… UC0wMDAyMTQ2OmJ… blca_plasmacytoid… P-00021… P-0002146 blca_p…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> c19DX0NBVVdUN1… cF9DX0NBVVdUNzp… prad_msk_2019_str… s_C_CAU… p_C_CAUW… prad_m…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 38 more variables: site1EntrezGeneId &lt;int&gt;, site1HugoSymbol &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1EnsemblTranscriptId &lt;chr&gt;, site1Chromosome &lt;chr&gt;, site1Position &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1Contig &lt;chr&gt;, site1Region &lt;chr&gt;, site1RegionNumber &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site1Description &lt;chr&gt;, site2EntrezGeneId &lt;int&gt;, site2HugoSymbol &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2EnsemblTranscriptId &lt;chr&gt;, site2Chromosome &lt;chr&gt;, site2Position &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2Contig &lt;chr&gt;, site2Region &lt;chr&gt;, site2RegionNumber &lt;int&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   site2Description &lt;chr&gt;, site2EffectOnFrame &lt;chr&gt;, ncbiBuild &lt;chr&gt;, …</span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

