<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • cbioportalR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="https://karissawhiting.github.io/cbioportalR/logo.png"><meta name="robots" content="noindex"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZFP6P2RCRE"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-ZFP6P2RCRE');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">cbioportalR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/overview-of-workflow.html">Overview of Data Retrieval Workflow</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/karissawhiting/cbioportalR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/karissawhiting/cbioportalR/blob/main/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="development version" id="cbioportalr-development-version">cbioportalR (development version)<a class="anchor" aria-label="anchor" href="#cbioportalr-development-version"></a></h2>
<ul><li>Fixed bug in <code><a href="../reference/available_studies.html">available_studies()</a></code> that arose with underlying API change from V2 to V3.</li>
<li>Added <code><a href="../reference/lookup_id.html">lookup_id()</a></code> function to return all samples that exist across cBioPortal for a given patient ID.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.1.0" id="cbioportalr-110">cbioportalR 1.1.0<a class="anchor" aria-label="anchor" href="#cbioportalr-110"></a></h2><p class="text-muted">CRAN release: 2023-10-05</p>
<ul><li>Added <code><a href="../reference/available_sample_lists.html">available_sample_lists()</a></code> function which returns all available sample list IDs for a given study ID</li>
<li>Added <code>sample_list_id</code> argument to <code><a href="../reference/available_samples.html">available_samples()</a></code> which returns all samples IDs in specific sample list within a study (<a href="https://github.com/karissawhiting/cbioportalR/issues/53" class="external-link">#53</a>).</li>
<li>Fixed {cli} errors to make package compatible with {cli} ≥ v3.4.1.</li>
<li>Added CNA segmentation retrieval endpoint accessible via <code><a href="../reference/get_segments_by_sample.html">get_segments_by_sample()</a></code> and <code><a href="../reference/get_segments_by_study.html">get_segments_by_study()</a></code>). You can use <code>get_genomics_by_*(return_segments = TRUE)</code> as well to access this data.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.0.1" id="cbioportalr-101">cbioportalR 1.0.1<a class="anchor" aria-label="anchor" href="#cbioportalr-101"></a></h2><p class="text-muted">CRAN release: 2022-08-29</p>
<p>This version makes the package compatible with <a href="https://github.com/cBioPortal/cbioportal/releases/tag/v5.0.0" class="external-link">cBioPortal v5.0</a>. Main updates include:</p>
<ul><li>Fusion API endpoint URLs updated to use <code>structural-variant</code> instead of <code>fusion</code> in specification.</li>
<li>
<code><a href="../reference/get_fusions_by_sample.html">get_structural_variants_by_sample()</a></code> and <code><a href="../reference/get_fusions_by_study.html">get_structural_variants_by_study()</a></code> added as aliases for <code><a href="../reference/get_fusions_by_sample.html">get_fusions_by_sample()</a></code> and <code><a href="../reference/get_fusions_by_study.html">get_fusions_by_study()</a></code>. These functions return the same results as their fusion counterparts and both names will be supported.</li>
<li>When functions include the <code>data_type</code> argument, <code>structural_variant</code> is now available as option. This will return the same results as <code>fusion</code>.</li>
<li>Output of <code><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample()</a></code> and <code><a href="../reference/get_genetics_by_study.html">get_genetics_by_study()</a></code> functions are now named lists with names: <code>mutation</code>, <code>cna</code>, and <code>structural_variant</code> (changed from <code>fusion</code>) to be consistent with cBioPortal v5.0 naming conventions.</li>
<li>Tests and examples are no longer run on CRAN to protect from future changes to internal database. They are still run in Github Actions and examples are run on website.</li>
<li>Some examples adjusted to use a new cBioPortal profile name, as needed (e.g. <code>prad_msk_2019_structural_variants</code> -&gt; <code>prad_msk_2019_structural_variants</code>)</li>
<li>bug fix in <code>add_hugo</code> argument of <code>get_*_by_sample()</code> functions.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.0.0" id="cbioportalr-100">cbioportalR 1.0.0<a class="anchor" aria-label="anchor" href="#cbioportalr-100"></a></h2><p class="text-muted">CRAN release: 2022-08-10</p>
<p>This is the first release submitted to CRAN, and includes all updates (listed below under versions 0.2.0 and 0.2.1) made since last Github release (0.1.0). Package has been overhauled since first release (0.1.0), and code will not be backwards compatible with that version.</p>
<div class="section level4">
<h4 id="summary-of-updates-since-1-0-0">Summary of Updates (since 0.1.0)<a class="anchor" aria-label="anchor" href="#summary-of-updates-since-1-0-0"></a></h4>
<ul><li>Authentication schema has been updated. Package-level environmental variable can now be set that specifies login credentials for that session</li>
<li>Functionality to pull fusion data has been added</li>
<li>Users can pull data by study ID, sample ID, and patient ID</li>
<li>Update <code>get_genomics_by_sample()</code> and related functions to pull all gene data available for select samples instead of pulling data for IMPACT genes only by default</li>
<li>Allow users to pass Hugo Symbols or Panel IDs to all <code>get_genomics_by_sample()</code> and related functions.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.1" id="cbioportalr-021">cbioportalR 0.2.1<a class="anchor" aria-label="anchor" href="#cbioportalr-021"></a></h2>
<ul><li>Update <code>get_genomics_by_sample()</code> and related functions to pull all gene data available for selected sample by default instead of pulling data for IMPACT genes only by default. This will change default results when user does not specify genes, and could return more results than previously as they will include results for non IMPACT gene mutations, CNA or fusions, if available.</li>
<li>Allow users to pass Hugo Symbols to all <code>get_genomics_by_sample()</code> and related functions (previously had to specify by Entrez Gene ID or <code>NULL</code>) (<a href="https://github.com/karissawhiting/cbioportalR/issues/33" class="external-link">#33</a>)</li>
<li>Allow users to pass gene panels to all <code>get_genomics_by_sample()</code> and related functions with new <code>panel</code> argument. Previously could only specify <code>genes</code> by specified sets of gene IDs. (<a href="https://github.com/karissawhiting/cbioportalR/issues/15" class="external-link">#15</a>)</li>
<li>Update functions that take sample ID-study ID or patient ID-study ID pairs to be less stringent in what data frame columns names they accept. Previously these functions only accepted snake case (e.g. <code>sample_id</code>, <code>study_id</code>), but now accept any capitalization/delimiters (e.g. <code>sampleID</code>, <code>sample id</code>) (<a href="https://github.com/karissawhiting/cbioportalR/issues/16" class="external-link">#16</a>)</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.0" id="cbioportalr-020">cbioportalR 0.2.0<a class="anchor" aria-label="anchor" href="#cbioportalr-020"></a></h2>
<ul><li>Update internals of most functions with the exception of <code><a href="../reference/cbp_api.html">cbp_api()</a></code>
</li>
<li>Genomic data is all pulled by the common internal functions <code><a href="../reference/dot-get_data_by_study.html">.get_data_by_study()</a></code> and <code><a href="../reference/dot-get_data_by_sample.html">.get_data_by_sample()</a></code> with wrappers for each data type</li>
<li>Updated function names to specify if user is pulling by study or sample ID</li>
<li>Authentication schema was updated. Package level environment variable now set instead of global variable for base URL specification</li>
<li>Updates to all documentation</li>
<li>Added extensive error messaging using {cli}</li>
<li>Added unit tests</li>
<li>Fixed fusion functionality. User can now pull fusion data by study ID or sample ID</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.0" id="cbioportalr-010">cbioportalR 0.1.0<a class="anchor" aria-label="anchor" href="#cbioportalr-010"></a></h2>
<ul><li>First release - basic API wrapping functionality only</li>
</ul></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

